nmtwizard/preprocess/prepoperator.py

```diff
@@ -412,17 +413,53 @@ def _preprocess(self, tu_batch, **kwargs):
         # TU operator applies an action to each tu.
         # The action yields zero, one or more element for the new list
         tu_list, meta_batch = tu_batch
-        tu_list = list(
-            chain.from_iterable(
-                self._preprocess_tu(tu, meta_batch, **kwargs) for tu in tu_list
-            )
-        )
-        return tu_list, meta_batch
+        new_tu_list = []
+        for tu in tu_list:
+            tu_src_detok = tu.src_detok
+            tu_tgt_detok = tu.tgt_detok
+            tu_fuzzy_tgt_detok = tu.get_tgt_detok("fuzzy")
```

Comment:
To avoid unecessary detokenization, this should only be run if `logger.isEnabledFor(log_level)` is `True`.

Reply:
Thanks, done.